osfoffline/filesystem_manager/osf_event_handler.py

```diff
@@ -203,18 +204,18 @@ def on_deleted(self, event):
         # get item
         item = self._get_item_by_path(src_path)
 
-        # put item in delete state
-        item.locally_deleted = True
-
-        # nodes cannot be deleted online. THUS, delete it inside database. It will be recreated locally.
-        if isinstance(item, Node):
-            session.delete(item)
-            save(session)
-            return
-
-        save(session, item)
-
-        logging.info('{} set to be deleted'.format(src_path.full_path))
+        # put item in delete state after waiting a second and
+        # checking to make sure the file was actually deleted
+        yield from asyncio.sleep(1)
+        if not os.path.exists(item.path):
+            item.locally_deleted = True
+            # nodes cannot be deleted online. THUS, delete it inside database. It will be recreated locally.
+            if isinstance(item, Node):
+                session.delete(item)
+                save(session)
```

Comment:
Please wrap all DB save calls in a try/except block, to avoid an unhandled DB exception.

(as we saw yesterday, data corruption does happen- we don't want it to hang the client)